src/main/java/de/tum/in/www1/artemis/service/iris/IrisSettingsService.java

```diff
@@ -0,0 +1,400 @@
+     * @param secondSubSettingsAreOptional Whether the exercise settings are optional or not
+     * @param reduced                      Whether only the enabled flag should be combined or all settings
+     * @return The combined sub-settings
+     */
+    private IrisSubSettings combineSubSettings(IrisSubSettings subSettings1, IrisSubSettings subSettings2, boolean secondSubSettingsAreOptional, boolean reduced) {
+        if (secondSubSettingsAreOptional && subSettings2 == null) {
+            return subSettings1;
+        }
+
+        var combinedSettings = new IrisSubSettings();
+
+        var enabled = subSettings2 != null && subSettings2.isEnabled() && subSettings1 != null && subSettings1.isEnabled()
+                && applicationContext.getEnvironment().acceptsProfiles(Profiles.of("iris-gpt3_5"));
+        combinedSettings.setEnabled(enabled);
+
+        if (!reduced) {
+            var preferredModel = subSettings2 != null && subSettings2.getPreferredModel() != null ? subSettings2.getPreferredModel()
+                    : subSettings1 != null ? subSettings1.getPreferredModel() : null;
+            combinedSettings.setPreferredModel(preferredModel);
+
+            var template = subSettings2 != null && subSettings2.getTemplate() != null ? subSettings2.getTemplate() : subSettings1 != null ? subSettings1.getTemplate() : null;
+            combinedSettings.setTemplate(template);
+        }
+
+        return combinedSettings;
+    }
+
+    /**
+     * Adds the default Iris settings to a course if they are not present yet.
+     *
+     * @param course The course to add the default Iris settings to
+     * @return The course with the default Iris settings
+     */
+    public Course addDefaultIrisSettingsTo(Course course) {
+        if (course.getIrisSettings() != null) {
+            return course;
+        }
+        course.setIrisSettings(createDefaultIrisSettings(true));
+        return courseRepository.save(course);
+    }
+
+    /**
+     * Adds the default Iris settings to a programming exercise if they are not present yet.
+     *
+     * @param programmingExercise The programming exercise to add the default Iris settings to
+     * @return The programming exercise with the default Iris settings
+     */
+    public ProgrammingExercise addDefaultIrisSettingsTo(ProgrammingExercise programmingExercise) {
+        if (programmingExercise.getIrisSettings() != null) {
+            return programmingExercise;
+        }
+        programmingExercise.setIrisSettings(createDefaultIrisSettings(false));
+        return programmingExerciseRepository.save(programmingExercise);
+    }
+
+    private IrisSettings createDefaultIrisSettings(boolean withOptionalSettings) {
+        var irisSettings = new IrisSettings();
+        irisSettings.setIrisChatSettings(createDefaultIrisSubSettings());
+        irisSettings.setIrisHestiaSettings(withOptionalSettings ? createDefaultIrisSubSettings() : null);
```

Comment:
could we make this an optional to avoid setting it to null?

Reply:
IntelliJ then tells me "'One To One' attribute type should not be 'Optional'", so I'd leave it as is for now.